### PR TITLE
Update plugin POM

### PR DESCRIPTION
**Pull request by basil**
Bumps the [plugin parent POM](https://github.com/jenkinsci/plugin-pom) from 3.21 to 3.48.

See [the changelog](https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md) and the full diff in [the compare view](https://github.com/jenkinsci/plugin-pom/compare/plugin-3.21...plugin-3.48).

### CODE REVIEW

**Comment by basil** [comment]
@dwnusbaum Gentle ping. No rush on this, but just wanted to make sure the first notification didn't slip by.

File: pom.xml [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
+        <version>3.50</version>
         <relativePath />
     </parent>
     <groupId>org.jenkins-ci.plugins.workflow</groupId>
     <artifactId>workflow-scm-step</artifactId>
     <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <name>Pipeline: SCM Step</name>
-    <url>https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+SCM+Step+Plugin</url>
+    <url>https://wiki.jenkins.io/display/JENKINS/Pipeline+SCM+Step+Plugin</url>
```
Comment (MarkEWaite) [inline pom.xml]: This seems like a great excuse to change from using the inadequate documentation on the wiki to the existing documentation in the README.  The README is easier to read, contains more information, and better supports the user.
```suggestion
    <url>https://github.com/jenkinsci/workflow-scm-step-plugin</url>
```

@dwnusbaum this does not provide a changelog, but I suspect you will want to transition to use release drafter to provide the release notes automatically.

**Comment by dwnusbaum** [comment]
@basil Thanks for the PR! I went ahead and merged #36 instead, since it went to a newer version of the parent pom. Like Mark mentioned, it would be best to not update the URL until the docs are migrated to GitHub, which I'll do whenever I release the next version unless someone else does it first, so I am going to go ahead and close this PR.

File: pom.xml (hunk 1 above)
Comment (dwnusbaum) [reply pom.xml]: Yes, I have been migrating the changelog/documentation for Pipeline plugins to GitHub whenever I need to release a plugin that has not yet been migrated.

**Pull request closed without merging**